No such luck:

```
src/bin/sage -t --long --warn-long 30.0 --random-seed=286735480429121101562228604801325644303 src/sage/misc/dev_tools.py
**********************************************************************
Error: Failed example:: Exception raised:
Traceback (most recent call last):
  File "/sage/src/sage/doctest/forker.py", line 733, in _run
    self.compile_and_execute(example, compiler, test.globs)
  File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
    exec(compiled, globs)
  File "<doctest sage.misc.dev_tools.find_objects_from_name[1]>", line 1, in <module>
    dt.find_objects_from_name('FareySymbol')                                  # needs sage.modular
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
    for smodule_name, smodule in sys.modules.items():
RuntimeError: dictionary changed size during iteration

    dt.find_objects_from_name('FareySymbol')                                  # needs sage.modular
Exception raised:
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 733, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.misc.dev_tools.find_objects_from_name[1]>", line 1, in <module>
        dt.find_objects_from_name('FareySymbol')                                  # needs sage.modular
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
        for smodule_name, smodule in sys.modules.items():
    RuntimeError: dictionary changed size during iteration
**********************************************************************
Error: Failed example:: Exception raised:
Traceback (most recent call last):
  File "/sage/src/sage/doctest/forker.py", line 733, in _run
    self.compile_and_execute(example, compiler, test.globs)
  File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
    exec(compiled, globs)
  File "<doctest sage.misc.dev_tools.find_objects_from_name[3]>", line 1, in <module>
    dt.find_objects_from_name('RR')
  File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
    for smodule_name, smodule in sys.modules.items():
RuntimeError: dictionary changed size during iteration

    dt.find_objects_from_name('RR')
Exception raised:
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 733, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.misc.dev_tools.find_objects_from_name[3]>", line 1, in <module>
        dt.find_objects_from_name('RR')
      File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
        for smodule_name, smodule in sys.modules.items():
    RuntimeError: dictionary changed size during iteration
**********************************************************************
Error: Failed example:: Exception raised:
Traceback (most recent call last):
  File "/sage/src/sage/doctest/forker.py", line 733, in _run
    self.compile_and_execute(example, compiler, test.globs)
  File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
    exec(compiled, globs)
  File "<doctest sage.misc.dev_tools.find_objects_from_name[7]>", line 1, in <module>
    objs = dt.find_objects_from_name('find_objects_from_name')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
    for smodule_name, smodule in sys.modules.items():
RuntimeError: dictionary changed size during iteration

    objs = dt.find_objects_from_name('find_objects_from_name')
Exception raised:
    Traceback (most recent call last):
        import_statements('Z')
      File "/sage/src/sage/misc/dev_tools.py", line 528, in import_statements
        modules.update(find_object_modules(o))
                       ^^^^^^^^^^^^^^^^^^^^^^
      File "/sage/src/sage/misc/sageinspect.py", line 2778, in find_object_modules
        for module_name, module in sys.modules.items():
    RuntimeError: dictionary changed size during iteration
**********************************************************************
Error: Failed example:: Got: 
Traceback (most recent call last):
  File "/sage/src/sage/doctest/forker.py", line 733, in _run
    self.compile_and_execute(example, compiler, test.globs)
  File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
    exec(compiled, globs)
  File "<doctest sage.misc.dev_tools.import_statements[17]>", line 1, in <module>
    import_statements('HeckeMonoid')
  File "/sage/src/sage/misc/dev_tools.py", line 506, in import_statements
    obj = find_objects_from_name(name, include_lazy_imports=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
    for smodule_name, smodule in sys.modules.items():
RuntimeError: dictionary changed size during iteration

    import_statements('HeckeMonoid')
Expected:
    Traceback (most recent call last):
    ...
    LookupError: no object named 'HeckeMonoid'
Got:
    <BLANKLINE>
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 733, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1157, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.misc.dev_tools.import_statements[17]>", line 1, in <module>
        import_statements('HeckeMonoid')
      File "/sage/src/sage/misc/dev_tools.py", line 506, in import_statements
        obj = find_objects_from_name(name, include_lazy_imports=True)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/sage/src/sage/misc/dev_tools.py", line 279, in find_objects_from_name
        for smodule_name, smodule in sys.modules.items():
    RuntimeError: dictionary changed size during iteration
**********************************************************************
2 items had failures:
   5 of  12 in sage.misc.dev_tools.find_objects_from_name
   4 of  38 in sage.misc.dev_tools.import_statements
    [60 tests, 9 failures, 3.16s wall]
```

Naturally it works fine when I re-run it on my machine.

--- a/src/sage/misc/dev_tools.py
+++ b/src/sage/misc/dev_tools.py
@@ -275,8 +275,12 @@ def find_objects_from_name(name, module_name=None, include_lazy_imports=False):
     """
     from sage.misc.lazy_import import LazyImport
 
+    # Create a copy to avoid errors if the sys.modules dict changes
+    # while we are iterating over it.
+    mods = sys.modules.copy()
+
     obj = []
-    for smodule_name, smodule in sys.modules.items():
+    for smodule_name, smodule in mods.items():
         if module_name and not smodule_name.startswith(module_name):
             continue
         if hasattr(smodule, '__dict__') and name in smodule.__dict__:

--- a/src/sage/rings/padics/padic_lattice_element.py
+++ b/src/sage/rings/padics/padic_lattice_element.py
@@ -5,23 +5,6 @@ AUTHOR:
 
 - Xavier Caruso (2018-02): initial version
 
-TESTS:
-
-We create some rings and run the test suite for them. We skip the Smith form
-tests because they take a few minutes as of mid 2018, see :issue:`25431`::
-
-    sage: R1 = ZpLC(2)
-    doctest:...: FutureWarning: This class/method/function is marked as experimental. It, its functionality or its interface might change without a formal deprecation.
-    See https://github.com/sagemath/sage/issues/23505 for details.
-    sage: R2 = ZpLF(2)
-    sage: R3 = QpLC(2)
-    sage: R4 = QpLF(2)
-
-    sage: # long time, needs sage.rings.padics
-    sage: TestSuite(R1).run(skip=['_test_teichmuller', '_test_matrix_smith'])
-    sage: TestSuite(R2).run(skip=['_test_teichmuller', '_test_matrix_smith'])
-    sage: TestSuite(R3).run(skip=['_test_teichmuller', '_test_matrix_smith'])
-    sage: TestSuite(R4).run(skip=['_test_teichmuller', '_test_matrix_smith'])
 """
 
 # ****************************************************************************

--- a/src/sage/rings/padics/padic_lattice_element_test.py
+++ b/src/sage/rings/padics/padic_lattice_element_test.py
@@ -1,27 +1,65 @@
 import pytest
-import warnings
 
 
-def test_padic_lattice_element():
+@pytest.fixture
+def R1():
+    from warnings import catch_warnings, filterwarnings
+    from sage.rings.padics.factory import ZpLC
+    with catch_warnings(category=FutureWarning):
+        filterwarnings("ignore", category=FutureWarning)
+        return ZpLC(2)
+
+
+@pytest.fixture
+def R2():
+    from warnings import catch_warnings, filterwarnings
+    from sage.rings.padics.factory import ZpLF
+    with catch_warnings(category=FutureWarning):
+        filterwarnings("ignore", category=FutureWarning)
+        return ZpLF(2)
+
+
+@pytest.fixture
+def R3():
+    from warnings import catch_warnings, filterwarnings
+    from sage.rings.padics.factory import QpLC
+    with catch_warnings(category=FutureWarning):
+        filterwarnings("ignore", category=FutureWarning)
+        return QpLC(2)
+
+
+@pytest.fixture
+def R4():
+    from warnings import catch_warnings, filterwarnings
+    from sage.rings.padics.factory import QpLF
+    with catch_warnings(category=FutureWarning):
+        filterwarnings("ignore", category=FutureWarning)
+        return QpLF(2)
+
+
+# Use strings for the fixture names here, and then later convert them
+# to the actual fixture objects using request.getfixturevalue(). This
+# is a workaround for being unable to pass fixtures directly as
+# parameters:
+#
+#   https://github.com/pytest-dev/pytest/issues/349
+#
+elements = ( "R1", "R2", "R3", "R4" )
+
+
+@pytest.mark.parametrize("e", elements)
+def test_padic_lattice_element(e, request):
     r"""
     Run the ``TestSuite()`` for some examples that previously
     lived in the TESTS:: block of the padic_lattice_element module.
     """
     from sage.misc.sage_unittest import TestSuite
-    from sage.rings.padics.factory import ZpLC, ZpLF, QpLC, QpLF
 
-    with warnings.catch_warnings(category=FutureWarning):
-        warnings.filterwarnings("ignore", category=FutureWarning)
-        # These all raise FutureWarnings
-        R1 = ZpLC(2)
-        R2 = ZpLF(2)
-        R3 = QpLC(2)
-        R4 = QpLF(2)
+    # Convert the string to a real fixture
+    e = request.getfixturevalue(e)
 
-    for R in (R1, R2, R3, R4):
-        # Only do a few runs, _test_matrix_smith() in particular is
-        # sloooooow.
-        TestSuite(R).run(verbose=True,
-                         raise_on_failure=True,
-                         skip="_test_teichmuller",
-                         max_runs=8)
+    # Only do a few runs, _test_matrix_smith() in particular is slow.
+    TestSuite(e).run(verbose=True,
+                     raise_on_failure=True,
+                     skip="_test_teichmuller",
+                     max_runs=8)

